### PR TITLE
scheduler: MaxPDVolumeCountChecker.predicate returns early if no new volume is needed

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind bug
/kind cleanup

**What this PR does / why we need it**:
scheduler: MaxPDVolumeCountChecker.predicate returns early if no new volume is needed

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -533,6 +533,12 @@ func (c *MaxPDVolumeCountChecker) predicate(pod *v1.Pod, meta PredicateMetadata,
 	}
 
 	numNewVolumes := len(newVolumes)
+
+	// quick return
+	if numNewVolumes == 0 {
```
Comment (wgliang) [inline pkg/scheduler/algorithm/predicates/predicates.go]: Delete comments or more detailed comments (returns early if no new volume is needed).

**Review by wgliang — CHANGES_REQUESTED** [review]
Others LGTM

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (SataQiu) [reply pkg/scheduler/algorithm/predicates/predicates.go]: Thanks @wgliang

**Comment by wgliang** [comment]
/lgtm
Thanks @SataQiu

**Comment by SataQiu** [comment]
Hi @k82cn 
Could you take a look?

**Comment by SataQiu** [comment]
Can anyone take a look? Need some feedback! @kubernetes/sig-scheduling-pr-reviews

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (ahg-g) [reply pkg/scheduler/algorithm/predicates/predicates.go]: MaxPDVolumeCountChecker is deprecated, there is no reason to change its behavior.

**Comment by wojtek-t** [comment]
Closing based on https://github.com/kubernetes/kubernetes/pull/82992#discussion_r335466035

**Pull request closed without merging**